src/middleware/redux.ts

```diff
@@ -3,25 +3,31 @@ import { NamedSet } from './devtools'
 
 type Write<T, U> = Omit<T, keyof U> & U
 
-type ReduxState<A> = {
-  dispatch: StoreRedux<A>['dispatch']
-}
+type Action = { type: unknown }
 
-type StoreRedux<A> = {
+type StoreRedux<A extends Action> = {
   dispatch: (a: A) => A
   dispatchFromDevtools: true
 }
 
-type WithRedux<S, A> = Write<S, StoreRedux<A>>
+type ReduxState<A extends Action> = {
+  dispatch: StoreRedux<A>['dispatch']
+}
+
+type WithRedux<S, A extends Action> = Write<S, StoreRedux<A>>
 
-type Redux = <T, A, Cms extends [StoreMutatorIdentifier, unknown][] = []>(
+type Redux = <
+  T,
+  A extends Action,
+  Cms extends [StoreMutatorIdentifier, unknown][] = []
+>(
   reducer: (state: T, action: A) => T,
   initialState: T
 ) => StateCreator<Write<T, ReduxState<A>>, Cms, [['zustand/redux', A]]>
 
 declare module '../vanilla' {
   interface StoreMutators<S, A> {
-    'zustand/redux': WithRedux<S, A>
+    'zustand/redux': WithRedux<S, A & Action>
```

Comment:
I wasn't so sure, but is this what you mean? @devanshj

Reply:
Yes.

Bonus if you use `Cast` instead. To see the difference write a store with redux middleware, however on the `dispatch` function, you'll see in the action parameter there's an extra `& Action`, but if you use `Cast` the action type will be as it is. If you want I can also explain how `Cast` works for the record.

Reply:
I think I know what Cast does, but I don't get how it improves tooltip. Can this be explained with TS playground?

Reply:
There you go https://tsplay.dev/m3XyjW

Reply:
Thanks. I understand this one, and why `Cast` is preferable.